examples/custom_window_frame/src/main.rs

```diff
@@ -104,6 +98,38 @@ fn custom_window_frame(
                 frame.drag_window();
             }
 
+            // Add the close button:
+            let close_response = ui.put(
+                Rect::from_min_size(rect.left_top(), Vec2::splat(height)),
+                Button::new(RichText::new("❌").size(height - 4.0)).frame(false),
+            );
+            if close_response.clicked() {
+                frame.close();
+            }
+
+            let minimized_response = ui.put(
+                Rect::from_min_size(
+                    rect.left_top() + vec2((height - 4.0) * 1.0, 0.0),
+                    Vec2::splat(height),
+                ),
+                Button::new(RichText::new("--").size(height - 4.0)).frame(false),
+            );
+            if minimized_response.clicked() {
+                frame.set_minimized(true);
+            }
+
+            let maximized_response = ui.put(
+                Rect::from_min_size(
+                    rect.left_top() + vec2((height - 4.0) * 2.0, 0.0),
+                    Vec2::splat(height),
+                ),
+                Button::new(RichText::new("O").size(height - 4.0)).frame(false),
+            );
+            if maximized_response.clicked() {
+                app.maximized = !app.maximized;
```

Comment:
it would be even better if we could add `maximized` to `eframe::WindowInfo`. You can fill it in with https://docs.rs/winit/0.27.5/winit/window/struct.Window.html#method.is_maximized